lienzo-core/src/main/java/com/ait/lienzo/client/core/types/JsLienzo.java

```diff
@@ -0,0 +1,269 @@
+        for (WiresShape shape : shapes) {
+            if (id.equals(shape.getID())) {
+                final JsWiresShape jsWiresShape = new JsWiresShape(shape);
+                return jsWiresShape;
+            }
+        }
+        return null;
+    }
+
+    public String getBackgroundColor(String UUID) {
+
+        if (UUID == null || "".equals(UUID)) {
+            return null;
+        }
+        JsWiresShape shape = getWiresShape(UUID);
+        if (shape == null) {
+            return null;
+        }
+        return shape.getBackgroundColor();
+    }
+
+    public void setBackgroundColor(String UUID, String backgroundColor) {
+
+        if (UUID == null || "".equals(UUID) || backgroundColor == null || "".equals(backgroundColor)) {
+            return;
+        }
+
+        JsWiresShape shape = getWiresShape(UUID);
+        if (shape == null) {
+            return;
+        }
+        shape.setBackgroundColor(backgroundColor);
+    }
+
+    public String getBorderColor(String UUID) {
+
+        if (UUID == null || "".equals(UUID)) {
+            return null;
+        }
+
+        JsWiresShape shape = getWiresShape(UUID);
+        if (shape == null) {
+            return null;
+        }
+        return shape.getBorderColor();
+    }
+
+    public void setBorderColor(String UUID, String borderColor) {
+
+        if (UUID == null || "".equals(UUID) || borderColor == null || "".equals(borderColor)) {
+            return;
+        }
+
+        JsWiresShape shape = getWiresShape(UUID);
+        if (shape == null) {
+            return;
+        }
+        shape.setBorderColor(borderColor);
+    }
+
+    public NFastArrayList<Double> getLocation(String UUID) {
+
+        if (UUID == null || "".equals(UUID)) {
+            return null;
+        }
+
+        JsWiresShape shape = getWiresShape(UUID);
+        if (shape == null) {
+            return null;
+        }
+
+        final Point2D location = shape.getLocationXY();
+        NFastArrayList<Double> locationArray = new NFastArrayList<>();
+        locationArray.add(location.getX());
+        locationArray.add(location.getY());
+        return locationArray;
+    }
+
+    public NFastArrayList<Double> getAbsoluteLocation(String UUID) {
+
+        if (UUID == null || "".equals(UUID)) {
+            return null;
+        }
+
+        JsWiresShape shape = getWiresShape(UUID);
+        if (shape == null) {
+            return null;
+        }
+        final Point2D location = shape.getAbsoluteLocation();
+        NFastArrayList<Double> locationArray = new NFastArrayList<>();
+        calculatePanelOffset(location);
+        locationArray.add(location.getX());
+        locationArray.add(location.getY());
+
+        return locationArray;
+    }
+
+    protected void calculatePanelOffset(Point2D location) {
+        final elemental2.dom.HTMLElement editorPanel = (elemental2.dom.HTMLElement) DomGlobal.document.getElementById("canvasPanel");
```

Comment:
I think we should refactor this selector based on a hardcoded id (`canvasPanel`), and try to use some actual canvas element reference, is that possible? I'm just trying to prevent further missmatches when code gets updated over time...

Reply:
Hi @romartin I tried to make it more generic but there are many canvas tags that I could not just reference one. See.
![image](https://user-images.githubusercontent.com/39567766/135671701-13ca448c-d365-41a3-8a67-9d1dc59ce0db.png)